File: Python/linear_search.py
import random


def linear_search(arr, target):
    for index, value in enumerate(arr):
        if value==target:
            return index

    return -1


if __name__ == '__main__':
    random.seed(0)

    target = 101
    arr = [random.randint(-100,100) for _ in range(10)] + [target]
    random.shuffle(arr)

    index = linear_search(arr, target)

    print('Target found at index: ', index)
    print('Real index?', target==arr[index])
    print(arr)

File: C/linear_search.c
#include <stdio.h>


int LinearSearch(int arr[], int length, int target){
    for(int i=0; i<length; ++i){
        if(arr[i]==target){
            return i;
        }
    }

    return -1;
}


void Print(int arr[], int length){
    for(int i=0; i<length; ++i){
        printf("%d ", arr[i]);
    }
    
    printf("\n");
}


int main()
{
    int arr[10] = {-2, 94, 7, -90, -34, 30, 24, 3, 100, -23};
    
    Print(arr, 10);
    int index = LinearSearch(arr, 10, 30);
    
    printf("Target at index: ");
    printf("%d\n", index);

    return 0;
}
